src/qz/common/CachedObject.java

```diff
@@ -0,0 +1,97 @@
+package qz.common;
+
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Supplier;
+
+/**
+ * A generic class that encapsulates an object for caching. The cached object
+ * will be refreshed automatically when accessed after its lifespan has expired.
+ *
+ * @param <T> The type of object to be cached.
+ */
+public class CachedObject<T> {
+    public static final long DEFAULT_LIFESPAN = 5000;
+    T cachedObject;
+    Supplier<T> supplier;
+    private long timestamp;
+    private long lifespan;
+
+    /**
+     * Creates a new CachedObject with a default lifespan of 5000 milliseconds
+     *
+     * @param supplier The function to pull new values from
+     */
+    public CachedObject(Supplier<T> supplier) {
+        this(supplier, DEFAULT_LIFESPAN);
+    }
+
+    /**
+     * Creates a new CachedObject
+     *
+     * @param supplier The function to pull new values from
+     * @param lifespan The lifespan of the cached object in milliseconds
```

Comment:
Hmm.... has this been tested?  You cannot safely compare or add `millis` and `nanos`.  If `nanos` is preferred (I think it should be), it should be converted.  1 millisecond = 1000000 nanoseconds, but to that point, why use nanos to begin with?

Relevant: https://stackoverflow.com/q/351565/3196753

Reply:
Nevermind, you changed it from [your gist](https://gist.github.com/Vzor-/aeb69492221b89961ef53fde2e56b9be#file-cachedobject-java-L23) and it's using `toMillis` now. 🍻